font/src/ft/mod.rs

```diff
@@ -108,8 +108,8 @@ impl Rasterize for FreeTypeRasterizer {
 
         // Fallback for bitmap fonts which do not provide underline metrics
         if underline_position == 0. {
-            underline_thickness = (descent / 5.).round();
-            underline_position = descent / 2.;
+            underline_thickness = (-descent / 5.).round().max(1.);
+            underline_position = descent / 2. - (underline_thickness / 2.).round();
```

Comment:
After some more thought, the width of the underline should already be considered outside of our font crate. This is because freetype specifies the position as distance between baseline and **center** of the position. You can find that by searching for `underline_position` in the [freetype docs](https://www.freetype.org/freetype2/docs/tutorial/step2.html).

So as far as I can tell, we probably don't need to make any changes in the `ft/mod.rs` other than changing it from `descent` to `-descent`. We also already do the `max(1.)` in the `alacritty/renderer/rects.rs` file, so if we do a `max(1.)` in the `ft/mod.rs` we should probably check if it can be removed from Alacritty code. If it can't be, then we might not want it in the freetype code either, since a thickness < 1. seems acceptable then.

Which leads me to other things that can be tested out to resolve this. Currently our renderer computes the underline position based on the metrics and also includes the thickness of the underline to position it, since the position is the offset from the center. This is done [here](https://github.com/alacritty/alacritty/blob/906f14b6608480b601d08825461bd1d8eb295cfc/alacritty/src/renderer/rects.rs#L89).

Since the position is calculated based on the center of the underline, we have to divide the thickness and raise the position by half of it. However since we divide by two, this can lead to fractions (like 1.0 / 2.0 = 0.5). In these cases we need to decide if we either want to raise it to the next pixel or lower it. Since we should strive to prevent collisions with the glyphs themselves and we already have an upper limit (baseline - thickness), we should take the approach to lower it when we're not sure, rather than raising it. Currently this is not the case.

This change should thus be worth testing:
```
let mut y = (baseline - position - height / 2.).ceil();
```